module04/03-Panda-Social-Network/solution.py

```diff
@@ -0,0 +1,95 @@
+class Panda:
+    def __init__(self, name, email, gender):
+        self._name = name
+        self._email = email
+        self._gender = gender
+        return
+
+    def name(self):
+        return self._name
+
+    def email(self):
+        return self._email
+
+    def gender(self):
+        return self._gender
+
+    def isMale(self):
+        return self._gender == "male"
+
+    def isFemale(self):
+        return not self.isMale()
+
+    def __eq__(self, other):
+        result = self._name == other.name() and \
+                 self._email == other.email() and \
+                 self._gender == other.gender()
+
+        return result
+
+    def __hash__(self):
+        s = self.__str__()
+        h = hash(s)
+        h2 = hash((self.name(), self.email(), self.gender()))
+        return h
+
+    def __str__(self):
+        return f'Panda: ({self.name()}, {self.email()}, {self.gender()})'
+
+
+class PandaSocialNetwork:
+    def __init__(self):
+        self.network = {}
+        self.friends = {}
+        return
+
+    def add_panda(self, panda):
+        if self.has_panda(panda):
+            raise ValueError(f'I\'s already added {str(panda)} to the network.')
+
+        key = panda.__hash__()
+        self.network[key] = {key: panda}
+
+        return
+
+    def has_panda(self, panda):
+        key = panda.__hash__()
```

Comment:
You can simply use `hash(panda)` no actual need to use the system-defined one of the class that one is not meant to call.